calyx/src/passes/compile_sync.rs

```diff
@@ -48,11 +48,13 @@ impl Named for CompileSync {
 
 fn count_barriers(s: &ir::Control, count: &mut HashSet<u64>) {
     match s {
+        ir::Control::Empty(_) => {
```

Comment:
I would add an explicit case for `ir::Enable` and throw and error if you see a `@sync` on an invoke or an enable.